app/src/main/java/ch/epfl/sdp/peakar/points/POICache.java

```diff
@@ -0,0 +1,155 @@
+     * @param context context
+     */
+    private static void saveJson(String serializedObject,Context context){
+        File cacheDir = context.getCacheDir();
+        File outputFile = new File(cacheDir,CACHE_FILE_NAME);
+        try{
+            FileWriter writer = new FileWriter(outputFile);
+            writer.append(serializedObject);
+            writer.flush();
+            writer.close();
+        }
+        catch(IOException e){
+            Log.e("Exception", "File write failed cache POIPoints: " + e.toString());
+        }
+    }
+
+
+    /**
+     * Read text file and retrieve serialized
+     * @return serialized string
+     */
+    private static String readJSON(Context context){
+        File file = new File(context.getCacheDir(),CACHE_FILE_NAME);
+        StringBuilder fileContent = new StringBuilder();
+        try{
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(file));
+            String line;
+            while ((line = bufferedReader.readLine()) != null){
+                fileContent.append(line);
+                fileContent.append('\n');
+            }
+            bufferedReader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return fileContent.toString();
+    }
+
+    /**
+     * Retrieve POI data from cache and overwrite cachedPOIPoints and cachedBoundingBox
+     * with cache file values
+     */
+    private static void retrievePOIDataFromCache(Context context){
+        String fileContent = readJSON(context);
+        Gson gson = new Gson();
+        POICacheContent poiCacheContent = gson.fromJson(fileContent, POICacheContent.class);
+        cachedPOIPoints = poiCacheContent.getCachedPOIPoints();
+        cachedBoundingBox = poiCacheContent.getCachedBoundingBox();
+    }
+
+    /**
+     * Check if the user is located inside the middle part of the bounding box
+     * Using cached bounding box, a shrink-ed bounding box is computed.
+     * The methods then checks if the user position is somewhere inside this
+     * smaller bounding box
+     * @param userPoint user location
+     * @return true if the user is in the middle of the cached bounding box
+     *          false if not
+     */
+    public boolean isUserInBoundingBox(UserPoint userPoint,Context context){
+        //Get the bounding box from file if not already present
+        if(cachedBoundingBox == null)
+            retrievePOIDataFromCache(context);
+        //Decrease the size of the bounding box by half
+        //BoundingBox innerBox = cachedBoundingBox.increaseByScale(INNER_BOUNDING_BOX_SCALING_FACTOR);
+        return cachedBoundingBox.contains(userPoint.getLatitude(), userPoint.getLongitude());
+    }
+
+
+    /**
+     * Checks if cache file is present in device Cache folder
+     * @param context application context
+     * @return true if file is present. False if no file found
+     */
+    public boolean isCacheFilePresent(Context context){
```

Comment:
This function should in that case also be moved.

Reply:
No, this is really usefull to check if the data has to be download or retrieved from the cache.

Reply:
Yes I meant if all cache should be moved but since that is not the case the comment no longer applies.